### ISSUE TITLE
Images lost in IR deck

### ISSUE THREAD

**Issue by pkbs121** [luoliyan/incremental-reading-for-anki#27]
Hello, 
I am experiencing a problem where screen shots I have inserted into my IR cards are now appearing as small blue question marks. When I go to Tools>Check media, I have many files that are "in the media folder, but not used by cards". Also, I have many of the same files that are altered and appear in the "used on cards but missing from media folder". When I check my anki media collection on my computer, all my original screenshot files are there with no alterations. 

The media files that are "used on cards but missing from media folder" are variations of the original files in my media folder. 

Heres an example: 
"in media folder, not used by cards" - Screen Shot 2017-09-09 at 9.32.01 PM.png
but I also have 
"used on cards, missing from media file"- Screen%20Shot%202017-09-09%20at%202.43.19%20PM.png

I am not overly tech savy so Im not sure what to make of this. Given that I have had no issues using screen shots in my other cards/decks I have to assume its specific to IR?

I appreciate your time and effort on this app. 

I am using V 3.6.10, on anki 2.0.47 for mac.

**Comment by luoliyan** [comment]
I'll try to work out a solution, if IR is causing it. The problem is probably related to the `%20`, which is how browsers encode spaces in URLs. I suspect it isn't being unescaped.

I'll see what I can come up with.

**Comment by pkbs121** [comment]
Ok so I have tried a couple things. First, I am screen shotting out of PDFs (I have had no trouble with this in other addons like image occlusion) 

So I noticed if I had the PDF open on my browser that the URL would have the file name with 20%s in there so I thought maybe If i opened the PDFs on my computer and screen shotted from there i wouldn't have the same problem. But, It seems that the images are still dropping from the cards. A weird thing is that if its the first time viewing the IR card, the images will be there, but when I reschedule them the next time I see them they will be gone frequently. 

Also for example today. I was reading an IR card and the image was there. I extracted part of the text for a cloze deletion and when I came back to the IR card the image had been replaced by a small blue question mark.

**Comment by pkbs121** [comment]
Yeah, the images are there and as soon as I take a close deletion from the source IR card then they disappear

**Comment by aleksejrs** [comment]
https://github.com/luoliyan/incremental-reading-for-anki/issues/17

**Comment by luoliyan** [comment]
I don't think this is related to #17, but we will find out when I fix that one.

@pkbs121 Can you try the following:
1. Open one of those problem cards for editing
2. Click in the Text field
3. Open the HTML editor (<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>X</kbd>)
4. Replace each `%20` with a space
5. Close the editor

Does the image then display properly?

**Comment by pkbs121** [comment]
Yes. If I go in and remove the %20 manually the images will come back. But theres a couple things. As seen below, its not always just %20 that is put in between the spaces (I dont know if this is relevant). Also, I went in manually and removed the %252520 listed below.. The images reappeared. Then I took a cloze deletion from the card and the images once again disappeared. Upon checking the HTML, the spaces had then been replaced by %20 as seen below in the second example. 

```
Three patients were
not seen for follow-up 1 week later. Of&nbsp;</div><div><img src="Screen%252520Shot%2525202017-09-09%252520at%2525205.00.56%252520PM.png" /><img src="Screen%252520Shot%2525202017-09-09%252520at%2525205.01.04%252520PM.png" /></div><div>those 3 patients lost to follow-up, 2 of
them noted an immediate reduction in
pain greater than 2 points on the NPRS
postmanipulation.
```

```
Three patients were
not seen for follow-up 1 week later. Of&nbsp;</div><div><img src="Screen%20Shot%202017-09-09%20at%205.00.56%20PM.png" /><img src="Screen%20Shot%202017-09-09%20at%205.01.04%20PM.png" /></div><div>those 3 patients lost to follow-up, 2 of
them noted an immediate reduction in
pain greater than 2 points on the NPRS
postmanipulation.
```

**Comment by luoliyan** [comment]
I've made a change that should prevent this from happening. However, it won't affect existing cards - only new extracts.

Let me know if it solves the problem.

**Comment by pkbs121** [comment]
Ok so the file names are no longer being altered but heres whats happening. The images will be there, I make a cloze deletion and they disappear. I open up the HTML and all the image file names are correct. If I do not change anything and simply close the HTML editor, the images will once again reappear.

**Comment by luoliyan** [comment]
I'm really not sure if this is some weirdness with the toolkit or not. I'll keep working on it, but I'm a little stumped by it at the moment.
